### PR TITLE
Basic Go Programs - Day 4

### PR DESCRIPTION
**Pull request by mrniranjan**
Example programs on Functions, Arrays,
slices, Constants, UTF-8

Signed-off-by: Niranjan Mallapadi <niranjan@ashoo.in>

### CODE REVIEW

File: basics/a77.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+const fiveSeconds = 5 * time.Second
+
+func main() {
+	now := time.Now()
+	FiveNanoSeconds1 := now.Add(-5) 
+	FiveNanoSeconds2 := now.Add(-fiveSeconds)
+	// if we declare integer -5
+	myvar := -5
+	FiveNanoSeconds3 := now.Add(myvar) // will give compiler erro
```
Comment (gsr-shanks) [inline basics/a77.go:18]: ./a77.go:18:29: cannot use myvar (type int) as type time.Duration in argument to now.Add

Comment (gsr-shanks) [reply basics/a77.go:18]: FiveNanoSeconds3 is unused

Comment (gsr-shanks) [inline basics/a77.go]: FiveNanoSeconds3 is unused as well.

File: basics/a78.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+package main
+
+import (
+	"fmt"
+)
+
+const myNum = 123456789
+
+func main() {
+	start_num  := 18
```
Comment (gsr-shanks) [inline basics/a78.go]: 'Use camelcase instead of snake case' is what the IDE says :)

File: basics/a80.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// strings and unicode 
+package main
+
+import (
+	"fmt"
+	"unicode/utf8"
+)
+
+func main() {
+	cz_city := "Břno"
```
Comment (gsr-shanks) [inline basics/a80.go]: Use camelcase instead of snake case

File: basics/a87.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	fmt.Println(str)
+	input := bufio.NewScanner(os.Stdin)
+	for input.Scan() {
+		if input.Text() == "" {
+			break
+		} else {
+			name = input.Text()
+		}
+	}
+	fmt.Println(name)
```
Comment (gsr-shanks) [inline basics/a87.go:24]: This is not printed. The terminal just hangs in there.

File: basics/a90.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
+/* Another way to initialize maps with some data*/
+package main
+
+import (
+	"fmt"
+)
+
+func main() {
+	prime_numbers := map[int]int {
```
Comment (gsr-shanks) [inline basics/a90.go]: Use camelcase instead of snake case

File: basics/a97.go [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+func main() {
+	num := []int{1,2,3,4,5,6,7,8,9}
+	checkEven(num)
+}
+func checkEven(arr []int) {
+	for i:=0; i < len(arr); i++ {
+		if arr[i] % 2 == 0 {
+			fmt.Printf("%d ", arr[i])
+		}
+	}
```
Comment (gsr-shanks) [inline basics/a97.go:18]: Need a fmt.Println() here. Else % gets printed.

File: basics/a98.go [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+import "fmt"
+
+func main() {
+	numbers := []int{4,8,9,15,23,12}
+	a := GreatNum(numbers)
+	fmt.Printf("%d is bigger amount %d\n", a, numbers)
+}
+// returns greatest number from the array given
+func GreatNum(arr []int) int {
+	cur_great := arr[0] // store the first number as current greatest
```
Comment (gsr-shanks) [inline basics/a98.go]: Use camelcase instead of snake case

**Comment by gsr-shanks** [comment]
Reviewed. LGTM.

File: basics/a87.go (hunk 4 above)
Comment (mrniranjan) [reply basics/a87.go:24]: You will have to do ctrl-d to exit the terminal

Comment (mrniranjan) [reply basics/a87.go:24]: You will have to do ctrl-d to exit the terminal

**Review by mrniranjan — COMMENTED** [review]
Please check comment

**Pull request merged**